pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -308,15 +296,47 @@ func getPinnedLibraries(registry string) []libInfo {
 	return res
 }
 
-// getLibrariesLanguageDetection runs process language auto-detection and returns languages to inject for APM Instrumentation.
-// The languages information is available in workloadmeta-store and attached on the pod's owner.
-func (w *Webhook) getLibrariesLanguageDetection(pod *corev1.Pod) []libInfo {
-	if config.Datadog().GetBool("admission_controller.auto_instrumentation.inject_auto_detected_libraries") {
-		// Use libraries returned by language detection for APM Instrumentation
-		return w.getAutoDetectedLibraries(pod)
-	}
+type libInfoLanguageDetection struct {
+	libs    []libInfo
+	enabled bool
 
-	return nil
+	injected bool
+}
+
+func (l *libInfoLanguageDetection) podMutator(v version) podMutator {
+	return podMutatorFunc(func(pod *corev1.Pod) error {
+		if l == nil || len(l.libs) == 0 || !v.usesInjector() || l.injected {
+			return nil
+		}
+
+		var libs []string
+		for _, lib := range l.libs {
+			libs = append(libs, string(lib.lang))
+		}
+
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGES_DETECTED",
+			Value: strings.Join(libs, ","),
+		})
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGE_DETECTION_ENABLED",
```

Comment:
There is a distinction between "enabling language detection" and "enabling injection based on language detection". 

enabling language detection means running the process check/collector to detect processes and languages. This is controlled by `datadog.apm.instrumentation.language_detection.enabled` helm config flag. 

enabling injection based on language detection means using the results of auto-detection to inject only the needed libraries in the admission controller. This is controlled by `admission_controller.auto_instrumentation.inject_auto_detected_libraries` 

Given this, it is important to be clear on what we are passing down to the injector. Do we need to indicate both? If yes, the code should be adapted to do so. If not, which one is important to be passed down? 

With the current implementation in this PR, I see that we are indicating if **injection based on language detection** is enabled or not by embedding a boolean flag as an env var `DD_INSTRUMENTATION_LANGUAGE_DETECTION_ENABLED`. But the env var might be misleading and fool us to think that it indicates whether or not language detection is enabled.

Reply:
A small indication, to check if language detection is enabled or not, you can assert that `language_detection.enabled` and `language_detection.reporting.enabled` are both set to `true`

Reply:
I'm changing this to `DD_INSTRUMENTATION_LANGUAGE_DETECTION_INJECTION_ENABLED` so it's clearer